src/components/shortcircuit-analysis-result.js

```diff
@@ -4,17 +4,93 @@
  * License, v. 2.0. If a copy of the MPL was not distributed with this
  * file, You can obtain one at http://mozilla.org/MPL/2.0/.
  */
-import React from 'react';
+import React, { useCallback, useMemo, useRef } from 'react';
 import PropTypes from 'prop-types';
 import { useIntl } from 'react-intl';
 import { useSelector } from 'react-redux';
-import VirtualizedTable from './util/virtualized-table';
+import { AgGridReact } from 'ag-grid-react';
+import { makeStyles, useTheme } from '@mui/styles';
+import 'ag-grid-community/styles/ag-grid.css';
+import 'ag-grid-community/styles/ag-theme-alpine.css';
+import clsx from 'clsx';
+import { groupPostSort } from './util/sort-functions';
+
+const useStyles = makeStyles((theme) => ({
+    grid: {
+        width: 'auto',
+        height: '100%',
+        position: 'relative',
+    },
+}));
 
 const ShortCircuitAnalysisResult = ({ result }) => {
     const intl = useIntl();
+    const gridRef = useRef();
 
     const shortCircuitNotif = useSelector((state) => state.shortCircuitNotif);
 
+    const theme = useTheme();
+    const GRID_PREFIX = 'grid.';
```

Comment:
To refactor with
src/components/spreadsheet/equipment-table.js:const GRID_PREFIX = 'grid.';